### DISCUSSION TITLE
[3.0.6 & 3.0.7] Font sizes larger than 2xl not applied/recognized in web app built with VueJS 3.0

### DISCUSSION THREAD

**Discussion by eugenevk** [tailwindlabs/tailwindcss#6668] [Help]
**What version of Tailwind CSS are you using?**
v3.0.6/v3.0.7

**What build tool (or framework if it abstracts the build tool) are you using?**
VueJS 3.0, Composition API

**What version of Node.js are you using?**
v.14.15.4

**What browser are you using?**
Brave, Edge, Chrome (all don't work)

**What operating system are you using?**
Windows

**Describe your issue**
Font-sizes text-3xl, text-4xl, ..., text-9xl are not applied. Have tested them all. Font-size text-xl and text-2xl do work properly.
Before upgrading to tailwind v3 it used to work.
Visual Studio code still does recognize 3xl, ... , 9xl, and even shows the size settings properly, but in my web app the size is not applied.

![image](https://user-images.githubusercontent.com/19785727/146923232-b61a4828-3263-43bb-9666-24b447e6d59b.png)

![image](https://user-images.githubusercontent.com/19785727/146923418-b8405acf-3270-41c9-81f1-5b49b1006326.png)

When I set the size to text-2xl I see this:

![image](https://user-images.githubusercontent.com/19785727/146923505-e3bb298d-4459-44b7-b808-73b07cbcb5a0.png)

![image](https://user-images.githubusercontent.com/19785727/146923551-66bfc976-5412-4766-ba8c-0285ef650695.png)

**Comment by adamwathan** [comment]
Hey! Can’t help without a reproduction.

**Reply by eugenevk** [reply]
Hi @adamwathan, I understand. Just don't know how to encapsulate the issue. My project is too large to easily replicate. It has integrations with Firebase and Algolia. Any suggestions?

**Reply by eugenevk** [reply]
@adamwathan, I notice a couple of things that you can hopefully clarify. I built a new app with "vue create <name>" and added TailwindCSS 3.0.7. I added the following tailwind.config.js file:
```
module.exports = {
  content: ["./src/**/*.{html,js}", "./src/**/*.vue"],
  theme: {
    extend: {},
  },
  plugins: [],
}
```
I added a styles.css file with solely this content:
```
@tailwind base;
@tailwind components;
@tailwind utilities;
```
I notice that if I use this command to generate the public styles.css file:
```
npx tailwindcss -i ./src/css/styles.css -o ./public/styles.css
```
that the text-3xl, etc. are correctly applied. However, if I don't use npx and just change the html file by adding something like class="text-3xl", it doesn't get applied in the browser. After I execute the npx command, the browser updates and shows the correct font size.
Why is this, because I am not changing the css file?

See second div in standard vue html example:
```
<template>
  <div class="hello">
    <div class="text-4xl">{{ msg }}</div>
    <p>
      For a guide and recipes on how to configure / customize this project,<br>
      check out the
      <a href="https://cli.vuejs.org" target="_blank" rel="noopener">vue-cli documentation</a>.
    </p>
    <h3>Installed CLI Plugins</h3>
    <ul>
      <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-router" target="_blank" rel="noopener">router</a></li>
    </ul>
    <h3>Essential Links</h3>
    <ul>
      <li><a href="https://vuejs.org" target="_blank" rel="noopener">Core Docs</a></li>
      <li><a href="https://forum.vuejs.org" target="_blank" rel="noopener">Forum</a></li>
      <li><a href="https://chat.vuejs.org" target="_blank" rel="noopener">Community Chat</a></li>
      <li><a href="https://twitter.com/vuejs" target="_blank" rel="noopener">Twitter</a></li>
      <li><a href="https://news.vuejs.org" target="_blank" rel="noopener">News</a></li>
    </ul>
    <h3>Ecosystem</h3>
    <ul>
      <li><a href="https://router.vuejs.org" target="_blank" rel="noopener">vue-router</a></li>
      <li><a href="https://vuex.vuejs.org" target="_blank" rel="noopener">vuex</a></li>
      <li><a href="https://github.com/vuejs/vue-devtools#vue-devtools" target="_blank" rel="noopener">vue-devtools</a></li>
      <li><a href="https://vue-loader.vuejs.org" target="_blank" rel="noopener">vue-loader</a></li>
      <li><a href="https://github.com/vuejs/awesome-vue" target="_blank" rel="noopener">awesome-vue</a></li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'HelloWorld',
  props: {
    msg: String
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
```

**Reply by eugenevk** [reply]
I tested this now also on my project and when using npx after a change, it works. Again, I fail to understand why this is needed. Can you please explain?

**Reply by adamwathan** [reply]
Can you create a new Vue project that demonstrates the same behavior and provide a link to that? Right now I have no idea how you've got it set up so it's impossible to help.

**Reply by eugenevk** [reply]
Can you recommend a playground where I can upload the project to for you?

**Reply by adamwathan** [reply]
You can create a public repository here on GitHub.

**Reply by eugenevk** [reply]
You can find it here: https://github.com/eugenevk/tailwindvue

**Reply by Uteev** [reply]
@eugenevk Maybe you didn't configure properly content section of tailwind config file. If tailwind doesn't know that some of your code uses text-3xl it will not create the neccessary css styles for it.

**Reply by Comi9** [reply]
This issue still happens with 3.0.7.

**Reply by Comi9** [reply]
Found out that if we use any other means of specifying the tw css classes, besides declaring them as strings (className="text-2xl"), using string interpolation, for example, will have TW preprocessor skip generating that class.

**Comment by Comi9** [comment]
I believe this was a Vue-related topic while my intervention addresses React. Sorries!

**Comment by welschmoor** [comment]
Same problem here. Using 3xl and 4xl makes font sizes be regular size (something like base or md). React 17.

**Reply by welschmoor** [reply]
Solution: under purge add a safelist in tailwind.config:
```

  purge: {
    content: [
      '...',
    ],
    options: {
      safelist: {
        standard: [
        'text-2xl',
        'text-3xl',
        'text-4xl',
        'text-5xl',
        'text-6xl',
        'sm:text-2xl',
        'sm:text-3xl',
        'sm:text-4xl',
        'sm:text-5xl',
        'sm:text-6xl',
        'md:text-2xl',
        'md:text-3xl',
        'md:text-4xl',
        'md:text-5xl',
        'md:text-6xl',
        'lg:text-2xl',
        'lg:text-3xl',
        'lg:text-4xl',
        'lg:text-5xl',
        'lg:text-6xl',
        ],
      },
    },

```

Under theme I have listed all font sizes extra. It could be redundant, but I have no cleaned the code up yet. This is React 17.

```
  theme: {
    fontSize: {
      'xs': '.75rem',
      'sm': '.875rem',
      'base': '1rem',
      'lg': '1.125rem',
      'xl': '1.25rem',
      '2xl': '1.5rem',
      '3xl': '1.875rem',
      '4xl': '2.25rem',
      '5xl': '3rem',
      '6xl': '4rem',
     },
     //...
  }
```

**Reply by rodrigo-arias** [reply]
Thanks this worked for me, here is more context: https://tailwindcss.com/docs/content-configuration#safelisting-classes

_One example of where this can be useful is if your site displays user-generated content and you want users to be able to use a constrained set of Tailwind classes in their content that might not exist in your own site’s source files._

**Reply by mananbordia** [reply]
Thanks, this works for me.

**Comment by lyqht** [comment]
Experiencing the same problem in React 18, using tailwind css v3.1.8

**Comment by Rub-s** [comment]
Experiencing the same problem in Vue 3 , tailwind css v3.1.8

**Comment by welschmoor** [comment]
I wrote the solution above, people. Read it. You need to prevent Tailwind from purging these classes. Mine was for V2. For V3 use the docs, if mine's not working.

**Comment by yogeshnogia** [comment]
I was having the issue, there was a problem with my content:[] tailwind.config.js file, components folder path was set wrong previously, I changed its path and it worked. 
Content property is mainly responsible for compiling the tailwind classes for project files
Make sure your content values contain the right path for the components you wanna render tailwindcss classes

Wrong path:

`    content: ["./src/**/*.{html,js}", './components/**/*.jsx'`
   
Right Path:

`    content: ["./src/**/*.{html,js}", './src/components/**/*.jsx'`

**Reply by hamasho** [reply]
This worked for me too. But I don't understand why.

Doesn't  `"./src/**/*.{html,js}"` include `'./src/components/**/*.jsx'`?

**Reply by wongjn** [reply]
No. `./src/**/*.{html,js}` expands to any `.html` or `.js` file in the `src` directory. If you wanted to include `.jsx` files in that single glob, you could add an additional entry to the brace expansion:

```diff
-./src/**/*.{html,js}
+./src/**/*.{html,js,jsx}
```

**Reply by TruePadawan** [reply]
This worked for me. I was using tailwind in a `layout` folder, which wasn't specified in the `tailwind.config.js` file

**Comment by AlienDwarf** [comment]
This is still an issue in Tailwind 3.3.3 and Vue3
`text-xl` is working as aspected but `text-2xl`, `text-3xl` and above is not applied.

**Reply by TruePadawan** [reply]
Confirm that the file you're using tailwind in is under the `content` property of your `tailwind.config.js` file

**Reply by lnoss** [reply]
Also, be sure that you are not overriding the values by customizing the default theme by mistake in `tailwind.config.js` under the `fontSize` key.
